### PR TITLE
Signature verification

### PR DESCRIPTION
**Pull request by ocallesp**
Implementation of signature and certificate validation.

### CODE REVIEW

File: src/clr_bundle_add.c [hunk 1]
```diff
@@ -53,6 +53,7 @@ static void print_help(const char *name)
 	printf("   -l, --list              List all available bundles for the current version of Clear Linux\n");
 	printf("   -x, --force             Attempt to proceed even if non-critical errors found\n");
 	printf("   -S, --statedir          Specify alternate swupd state directory\n");
+    printf("   -V, --verify-sign       Enable signature verification, default value is disabled\n");
```
Comment (ikeydoherty) [inline src/clr_bundle_add.c]: Indentation failure?

File: src/clr_bundle_add.c [hunk 2]
```diff
@@ -133,6 +135,9 @@ static bool parse_options(int argc, char **argv)
 		case 'x':
 			force = true;
 			break;
+        case 'V':
```
Comment (ikeydoherty) [inline src/clr_bundle_add.c]: Mixed tabs with spaces here  too

File: src/signature.c [hunk 3]
```diff
@@ -46,115 +46,297 @@
...
-		return false || (IMPL == IMPL_FORGIVE);
-	}
-	bio_data = BIO_new_file(data_filename, "r"); // i.e. fopen
-	if (bio_data == NULL) {
-		goto exit;
-	}
-	bio_sig = BIO_new_file(sig_filename, "r"); // i.e. fopen
-	if (bio_sig == NULL) {
-		goto exit;
+    if (!initialized) {
```
Comment (ikeydoherty) [inline src/signature.c]: This whole section now carries a new indentation (spaces) instead of tabs

**Comment by ikeydoherty** [comment]
I might suggest running `clang-format -i $(find . -name '*.[ch]')`

**Comment by phmccarty** [comment]
The access control lists for this repo were not set up the way I intended, so I force-pushed to master to remove this merge. Apologies for any inconvenience this may have caused.

**Pull request merged**